Спринт 8 - Макет и навигация

svetSanna opened the pull request:
Домашнее задание.
Остались вопросы. Подскажите пожалуйста:
1. Как в эмуляторе поменять язык? Отображается на английском языке. На моем телефоне отображается на русском.
2. Интересно, почему на моем телефоне отрабатывает "Ночная тема", хотя мы вроде не реализовали еще темы. Или это не ночная тема, а какой-то косяк? Но на моем телефоне в данном приложении вместо белого цвета черный, вместо черного - белый.
Спасибо

svetSanna committed:
Сделана верстка двух экранов: "Главный" и "Настройки"

svetSanna committed:
"Оживила" первый экран: при нажатии кнопок первого (главного) экрана выводятся соответствующие строчки (toast)

svetSanna committed:
"Оживила" первый экран: при нажатии кнопок первого (главного) экрана выводятся соответствующие строчки (toast)

svetSanna committed:
Добавлены две пустые активити для отображения Поиска и Медиатеки.
Реализован переход между активити по нажатию на кнопки, в том числе по кнопке "Назад" с активити настроек.

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,20 @@
+
+        button_settings_back.setOnClickListener {
+            val displayIntent = Intent(this, MainActivity::class.java)
+            startActivity(displayIntent)
```
Для перехода назад надо использовать не интент, а методы Activity finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на этот экран. На эту тему можно почитать про стек активити. Это будет критично в следующем спринте.

svetSanna replied:
Поняла. Исправила. Спасибо. В спринте не было, к сожалению, такой информации, поэтому сделала по аналогии с примером.

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,20 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_settings)
+
+        val button_settings_back = findViewById<Button>(R.id.button_settings_back)
```
Названия переменных и методов в Котлин и Джаве принято писать camelCase'ом, то есть buttonSettingsBack вместо button_settings_back. Айдишники в лейаутах да, часто пишутся в snake_case (хотя я и их предпочитаю писать в camelCase), но соответствующие им переменные надо переводить в camelCase.

svetSanna replied:
Исправила. Буду иметь ввиду

faritowich commented on app/src/main/res/drawable/poster.png:
Лишнее

svetSanna replied:
Убрала

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,127 @@
+            android:layout_marginLeft="12dp"
+              >
+        </TextView>
+    </LinearLayout>
```
Для тулбара можно также использовать один TextView и задавать иконку при помощи атрибута drawableStart или использовать androidx.appcompat.widget.Toolbar

svetSanna replied:
Не поняла, как можно сделать с помощью TextView. Вообще убрать кнопку и вместо нее сделать TextView? и обрабатывать нажатие на нее? А в приложениях обычно нажимаем ведь именно на стрелочку, а не на весь TextView. Или я что-то не поняла? Если не трудно, напишите пожалуйста, подробнее, как лучше сделать.
А Toolbar мы, наверное, будем скоро изучать.

faritowich replied:
Toolbar отдельно изучаться не будет, это просто один из многих компонентов UI в андроиде, про который полезно знать)
Насчет TextView - это нормальная практика, что кликабельна не только стрелочка, но и кнопка Назад, так что вполне можно сделать и текст, и стрелочку кликабельными.  Так что да, можно убрать кнопку. Вообще Button использовать для этого экрана я бы не стал даже для нижних кнопок в меню, а использовать бы Frame/LinearLayout со вложенными TextView и ImageView. Button лучше использовать для элементов, которые явно как кнопка выглядят, а у нас не тот случай

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,127 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="12dp"
+        android:layout_marginRight="14dp">
```
Повторяющиеся размеры

svetSanna replied:
Исправила

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,127 @@
+        </TextView>
+        <Button
+            android:layout_width="48dp"
+            android:layout_height="48dp"
```
Повторяющиеся размеры

svetSanna replied:
Исправила

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,127 @@
+            android:textSize="@dimen/text_size_menu_settings"
+            android:layout_gravity="center_vertical"
+            android:layout_marginBottom="21dp"
+            android:layout_marginTop="21dp">
```
layout_marginBottom + layout_marginBottom = layout_marginVertical

svetSanna replied:
Исправила

svetSanna committed:
Исправлены критические замечания ревьюера:
1. Убрана обводка вокруг кнопок на окне настроек
2. Исправлены все повторяющиеся числовые значения в разметке, точнее, вынесены в файл
Исправлены некритические замечания:
1. названия переменных изменены в соответствии со стилем camelCase
2. Для перехода назад теперь использую не интент, а метод onBackPressed()
3. Прочие исправления

svetSanna commented:
Все критические замечания исправила и постаралась исправить все некритические замечания. Про Toolbar не совсем поняла. В ответах на комментарий я написала свой вопрос. Буду рада получить ответ.

svetSanna committed:
После того как работа была принята ревьюером. Заменила на втором экране ("Настройки") Button на ImageView.

svetSanna committed:
Спринт 9. Добавила стили, шрифты.

svetSanna committed:
Спринт 9.
1. Сделала изменение цвета для switch (переключателя дневного и ночного режимов). Сделала одинаковые цвета switch для дня и ночи.
2. Реализовала кнопки экрана ActivitySettings. (переход в мессенджер, почту, браузер)

svetSanna committed:
Спринт 9. Домашнее задание 3. Сделала разметку для экрана "Поиск". Дневная тема.

svetSanna committed:
Спринт 9. Домашнее задание 3. Добавила ночную тему для экрана "Поиск".

svetSanna committed:
Спринт 9. Домашнее задание 3. Реализовала
1. работу EditText (Очистка содержимого по кнопочке),
2. возврат на предыдущее окно по стрелочке.

svetSanna committed:
Спринт 9. Домашнее задание 4. Реализовала сохранение текста поискового запроса в жизненном цикле SearchActivity.

svetSanna committed:
Спринт 9. Домашнее задание 4.Подправила в соответствии с требованиями. В частности, теперь при нажатии "Закрыть" в EditText скрывается клавиатура.
